packages/network/test/unit/agent_spec.ts

```diff
@@ -112,6 +154,20 @@ describe('lib/agent', function () {
               }
             }
 
+            if (testCase.httpsProxy) {
+              clientPkiCertificateStore.clear()
+              const certAndKey = createCertAndKey()
+              const pemCert = pki.certificateToPem(certAndKey[0])
+
+              this.clientCert = pemCert
+              const testCerts = new UrlPkiCertificates(`https://localhost`)
+
+              testCerts.pkiCertificates = new PkiCertificates()
+              testCerts.pkiCertificates.cert.push(pemCert)
+              testCerts.pkiCertificates.key.push(new PemKey(pki.privateKeyToPem(certAndKey[1]), undefined))
+              clientPkiCertificateStore.addPkiCertificatesForUrl(testCerts)
+            }
+
```

Comment:
hmm, we don't necessarily want to run all of our HTTPS tests with a client certificate, this should be moved to its own test/own set of tests